apps/src/lab2/views/dialogs/DialogManager.tsx

```diff
@@ -45,85 +37,39 @@ const DialogManager: React.FunctionComponent<DialogManagerProps> = ({
   children,
 }) => {
   const [openDialog, setOpenDialog] = useState<DialogType | null>(null);
-  const [dialogCallback, setDialogCallback] = useState<(() => void) | null>(
-    () => null
-  );
-  const [dialogTitle, setDialogTitle] = useState<string | undefined>(undefined);
-  const [dialogMessage, setDialogMessage] = useState<string | undefined>(
-    undefined
-  );
-  const [dialogConfirmText, setDialogConfirmText] = useState<
-    string | undefined
-  >(undefined);
+  const [dialogArgs, setDialogArgs] = useState<AnyDialogType>();
```

Comment:
Since each type of dialog has different args, we just read 'em in generically and pass along the object as appropriate.

Reply:
Not 100% sure if this would work but -

Since it looks like all Dialogs' props extend `GenericDialogProps`, we should be able to keep the type on `DialogViews` as `{[key in DialogType]: React.FunctionComponent<GenericDialogProps>}`? That way `DialogView` can be treated as a `FunctionComponent<GenericDialogProps>`. Not sure if that'll play nicely with the type of `dialogArgs` but we might be able to use some parameterized types to handle that too.

Reply:
Didn't work. Or at least I couldn't get it working.

Opting to just get it merged and we can circle back to it. I hate this stupid any. :angry: